### PR TITLE
Migrate to new testing APIs

### PR DESCRIPTION
**Pull request by simonihmig**
This is still WIP, as the support for the new testing APIs is not yet released in `ember-mocha` (using the master branch for now). So don't merge yet!

But all green it seems, so 🎉 /cc @Turbo87

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +42,7 @@
     "ember-cli-htmlbars": "^2.0.1",
     "ember-cli-htmlbars-inline-precompile": "^1.0.0",
     "ember-cli-inject-live-reload": "^1.4.1",
-    "ember-cli-mocha": "^0.14.4",
+    "ember-cli-mocha": "emberjs/ember-mocha#master",
```
Comment (Turbo87) [inline package.json]: interesting that this works... `ember-cli-mocha` vs `ember-mocha` 😉

Comment (simonihmig) [reply package.json]: Uh, good catch! 😬

Comment (simonihmig) [reply package.json]: Btw, do I still need `ember-cli-mocha` actually? Seems it only contains the blueprints nowadays, right?

Comment (Turbo87) [reply package.json]: https://github.com/ember-cli/ember-cli-eslint/pull/225 and code like that is the problem and the reason why we're keeping the "wrapper" for now

**Comment by simonihmig** [comment]
@andreasschacht this is now updated to the real release, and should be ready to be merged!

**Pull request merged**